### PR TITLE
Get Account at Block Height

### LINKED ISSUE #42: Implement `GetAccountAtBlockHeight`
### Issue To Be Solved

The `GetAccountAtBlockHeight` function is currently unimplemented in the emulator Access API. This function is useful for teams that need to query historical account information. 

https://github.com/onflow/flow-emulator/blob/33fb8fa82611592345ff6b9b023fb9b2705dfef8/server/backend/backend.go#L335

### Suggest A Solution

To implement this, we must first implement `emulator.Blockchain#GetAccountAtBlock`:

https://github.com/onflow/flow-emulator/blob/33fb8fa82611592345ff6b9b023fb9b2705dfef8/blockchain.go#L680-L682

It should be possible to create a veresion of `emulator.Blockchain#getAccount` thats support a custom height, rather than just the latest.

The key is the `b.storage.LedgerViewByHeight` function, which creates a view into the storage trie at a specific height.

https://github.com/onflow/flow-emulator/blob/33fb8fa82611592345ff6b9b023fb9b2705dfef8/blockchain.go#L662-L677

### PR DESCRIPTION
**Pull request by sideninja**
Closes #42 

## Description

Implemented fetching the account at specified block height and wrote tests.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by janezpodhostnik — APPROVED** [review]

File: blockchain.go [hunk 1]
```diff
@@ -675,44 +675,61 @@ func (b *Blockchain) GetAccount(address sdk.Address) (*sdk.Account, error) {
...
-	flowAddress := sdkconvert.SDKAddressToFlow(address)
-
-	account, err := b.getAccount(flowAddress)
+	latestBlock, err := b.GetLatestBlock()
 	if err != nil {
 		return nil, err
 	}
 
-	sdkAccount, err := sdkconvert.FlowAccountToSDK(*account)
+	return b.GetAccountAtBlock(address, latestBlock.Header.Height)
```
Comment (turbolent) [inline blockchain.go:683]: `GetAccount` locks and then `GetAccountAtBlock` locks again. See https://pkg.go.dev/sync#RWMutex.RLock.


Maybe replace with `getAccountAtBlock`, or, given that the body after the locking part at the beginning of this function is then equivalent to `getAccount`, just call `getAccount`.

**Review by turbolent — COMMENTED** [review]
Looks good, just one problem with the recursive locking

File: blockchain.go (hunk 1 above)
Comment (sideninja) [reply blockchain.go:683]: Oh right, thank you, nice catch.

**Review by turbolent — APPROVED** [review]
Nice!

**Pull request merged**